amazon/src/pages/Cart.page.tsx

```diff
@@ -1,9 +1,56 @@
 import HeaderComponent from '../features/products/components/Header.component';
+import PaymentGateway from '../features/products/components/Payment.component';
+import ProductComponent from '../features/products/components/Product.component';
+
+import { useAppSelector } from '../hooks/redux/hooks';
 
 const CartPage = () => {
+  const { cart, products } = useAppSelector((state) => state.product);
+
+  const totalQty = cart.reduce((acc, item) => acc + item.quantity, 0);
+  const totalPrice = cart.reduce(
+    (acc, item) => acc + item.quantity * item.price,
+    0
+  );
+
   return (
     <div>
       <HeaderComponent />
+      <div
+        style={{
+          display: 'flex',
+          flexWrap: 'wrap',
+          gap: '48px',
+          justifyContent: 'center',
+          alignItems: 'center',
+          marginTop: '48px',
+        }}
+      >
+        {products.length > 0 &&
+          products.map((product) => (
+            <ProductComponent key={product._id} product={product} />
+          ))}
+      </div>
+
+      <div style={{ width: '80%', margin: 'auto' }}>
+        <hr style={{ marginTop: '16px' }} />
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'flex-end',
+            fontSize: '20px',
+          }}
+        >
+          <span style={{ marginRight: '16px' }}>
+            Subtotal ({totalQty}) items:
+          </span>
+          <span style={{ fontWeight: 500, marginBottom: '48px' }}>
+            ${totalPrice.toFixed(2)}
+          </span>
+        </div>
+
+        {totalQty > 0 && <PaymentGateway />}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using cart.length for consistency</b></div>
<div id="fix">

The `PaymentGateway` component is conditionally rendered based on `totalQty > 0`. Consider using `cart.length > 0` instead for consistency and to avoid potential issues if `totalQty` is calculated incorrectly.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -52 +52 @@
 -        {totalQty > 0 && <PaymentGateway />}
 +        {cart.length > 0 && <PaymentGateway />}
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/amazon-clone/pull/9#issuecomment-2468531555>#8f0260</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged